build/commands/lib/config.js

```diff
@@ -246,7 +246,8 @@ Config.prototype.buildArgs = function () {
     target_cpu: this.targetArch,
     is_official_build: this.isOfficialBuild() && !this.isAsan(),
     is_debug: this.isDebug(),
-    dcheck_always_on: getNPMConfig(['dcheck_always_on']) || this.buildConfig !== 'Release',
+    dcheck_always_on: getNPMConfig(['dcheck_always_on']) ||
+      (this.buildConfig !== 'Release' && this.buildConfig !== 'Static'),
```

Comment:
nit `isComponentBuild()` but I think this setting should only be in here if `getNPMConfig(['dcheck_always_on'])` is set and otherwise we should use the upstream default. @iefremov I thought we were enabled DCHECK in PR builder?

Reply:
we can sort this out in a follow-up

Reply:
@bridiver 
DCHECKs are enabled in PR builder except for Windows, `dcheck_always_on` is set in DevOps repo.

Yeah, the best option is to do nothing if `getNPMConfig(['dcheck_always_on']) == undefined ` here and take it's value overwise.
But we can't do that without extra changes: `getNPMConfig` can't return values like 0, false, etc now.
In reason is the weird logic of  `getNPMConfig` : it takes variables for env that `npm run` sets, not from `npm config list` (that's why `-` in names, not `_` as in the configs).
I have a fix but better to do it in a separate issue.

Reply:
Replace to `isComponentBuild()`

Reply:
PR to solve the issues I'm taking about:
https://github.com/brave/brave-core/pull/12192